### ISSUE TITLE
`q_mean` variation in Linear Probing

### ISSUE THREAD

**Issue by rikardvinge** [DLR-MF-DAS/embed2scale-challenge-supplement#7]
We have identified and fixed a bug in the function which performs the repeated linear probings to gather statistics. 
Please resubmit your public submissions in order to reevaluate the Q_mean.

Pinging known participants:
@isaaccorley, @burakekim

**Comment by isaaccorley** [comment]
Hi @rikardvinge, given that there's a limit on submissions is it possible to reset this considering all of our team's submissions could potentially be invalid?

**Comment by rikardvinge** [comment]
@isaaccorley, let me see what we can do, do not resubmit quite yet, I'll give you a heads up

**Comment by rikardvinge** [comment]
@isaaccorley, I have now reevaluated your submissions, which should not count toward your total daily submissions. It might look like you've done many cancelled submissions, these comes from me, don't worry.
I reevaluated them in chronological order, so the submission shown on the two leaderboards correspond to your latest submission.

**Comment by burakekim** [comment]
That is great -- thank you for looping us in and reevaluating, @rikardvinge!

**Comment by isaaccorley** [comment]
@rikardvinge are you able to redo the baselines as well so we can get a feel for how much of an improvement is in our method?

**Comment by burakekim** [comment]
The very last two successful resubmissions appear to have the same submission link (under `Submitted File`) but very different `q_mean` scores. Could we possible look into this? 

Also, do you have any visibility into which exact submissions are the resubmitted ones -- there seems to be more successful resubmissions than original submissions. The timestamps, and perhaps download links, seem to be inaccurate and do not help with troubleshooting

**Comment by rikardvinge** [comment]
Thanks for highlighting the high variability, we are looking into it. We agree that it is too large.

Regarding the many successful resubmissions, this is likely my fault. I misunderstood how eval.ai worked when resubmitting and ran some submissions multiple times (still learning how to use the application :) ). The latest one is the one with the latest timestamp in the field `rerun_resumed_at` in the `Metadata file`. Sorry for the mess. In general, we strive for avoiding reevaluation as far as possible, partly due to it can cause these types of confusion and partly because there is an intended element of randomness in the evaluation which can hit both ways when resubmitting.

**Comment by burakekim** [comment]
That is perfectly fine -- we appreciate the great support! 

Re latest submissions: there does not seem to be a `Metadata file` field on my end, perhaps that is organizers-only?

**Comment by rikardvinge** [comment]
Regarding the high variability: We have a intentional component of randomness in the evaluation, for example we randomly sample the training and validation splits for an individual linear probe. This is to avoid overfitting to specifics in our evaluation.
However, you have a point that this randomness was too strong.
We have therefore tweaked the parameters to weaken the impact of this randomness.

**Comment by burakekim** [comment]
Great. Has the fix been reflected on the submission portal as well?

**Comment by burakekim** [comment]
Just submitted the same embedding twice, and it returned scores of -0.5 and 2.26 (the last two for reference). Not sure if the fix is already in place, but it seems randomness is still causing high variability

**Comment by rikardvinge** [comment]
Hi @burakekim, we have further analyzed the variability in the `q_mean` due to our statistical evaluation scheme. Here are the actions we implemented:
- We keep the calculations for `q_mean` as noted above. 
- However, we have increased the number of times we perform linear probing to gather statistics for each downstream task. This more or less corresponds to resubmitting several times and averaging the `q_mean` values.

In consequence, __@ all__, the evaluation time for a submission increased from the initial 15 seconds to around 2 minutes.
Regardless, we still expect some variability in `q_mean` on resubmission. This is the nature of the statistical evaluation we do in the linear probing of the embeddings. For the evaluation during the Test phase, we will further increase the statistical evaluation.

Thank you @burakekim and @isaaccorley for helping us fine-tune the competition!

**Comment by cmalbrec** [comment]
@burakekim , to close the loop on https://github.com/DLR-MF-DAS/embed2scale-challenge-supplement/issues/7#issuecomment-2725086523 : Yes, participants do not have access to the `Metadata file`.

**Comment by GustavMH** [comment]
I would appreciate any input and insight, into the following behavior of the `q_mean` score.

I ran some experiments to sanity check my `q_mean` results.
Run-to-run variance with the exact same embeddings is around 1, run 1: 7.84, run 2: 8.86.
Removing the mean and dividing the embeddings by 2 (a linear transform), the `q_mean` changes from 9.27 to 2.26.

This does not seem to down to run-to-run variance, the following questions arise:
- is the decoder network linear?
- is there a specific range of values that is accepted?
- if so, are the submissions normalized, or am I supposed to infer this range?

**Comment by rikardvinge** [comment]
Hi @GustavMH and **@ all**,

Your questions triggered another round of fine-tuning of the linear probing. Yes, the decoder network is linear. 

We realized that the range had an implicit effect on the training of the linear probe. Now, we introduced a normalization of the submissions. Therefore, we kindly advise all participants to resubmit.

**Comment by isaaccorley** [comment]
Hi @rikardvinge could you clarify what type of normalization you are using? E.g. L2, Standard norm using each vector statistics, standard norm of each column statistics independently?

**Comment by cmalbrec** [comment]
Hi @isaaccorley , we normalize by a global mean shift (computed from all embedding values) divided by the global standard deviation.
